src/ppl/src/HungarianMatching.cpp

```diff
@@ -223,9 +224,9 @@ void HungarianMatching::createMatrixForGroups()
 
       hungarian_matrix_[slot_index].resize(num_pin_groups_,
                                            std::numeric_limits<int>::max());
-      for (const std::vector<int>& io_group : pin_groups_) {
+      for (const auto& io_group : pin_groups_) {
```

Comment:
you can use a [structured binding](https://en.cppreference.com/w/cpp/language/structured_binding) here